### PR TITLE
framework/st_things : Add internal API st_things_stop.

### PR DESCRIPTION
**Pull request by Hyunjun85**
It will be able to support the restarting stack.

### CODE REVIEW

**Comment by seinfra** [comment]
Target : [3fc652ba05dc3b4d242a35e948194505a03a582b] - Code Rule Check (C++) OK.

File: framework/src/st_things/things_stack/things_stack.c [hunk 1]
```diff
@@ -88,6 +88,7 @@ things_reset_confirm_func_type g_confirm_reset_start = NULL;
 things_sec_otm_state_func_type g_otm_event_handler = NULL;
 
 static pthread_mutex_t g_things_stop_mutex = PTHREAD_MUTEX_INITIALIZER;
+static pthread_mutex_t m_thread_things_reset = PTHREAD_MUTEX_INITIALIZER;
```
Comment (JeongJunSik) [inline framework/src/st_things/things_stack/things_stack.c]: why you define this variable?

**Comment by seinfra** [comment]
Target : [a64ae08ed6b2a1fd658e6e615102deb6db29e263] - Code Rule Check (C++) OK.

**Comment by seinfra** [comment]
Target : [a64ae08ed6b2a1fd658e6e615102deb6db29e263] - Code Rule Check OK.

**Comment by seinfra** [comment]
Target : [edfac5a7b3b2c78266f185e22ba2094fe13f3d13] - Code Rule Check Result:
framework/src/st_things/things_stack/framework/things_data_manager.c:2007: ERROR: [SPC_M_SEP] space prohibited after that open parenthesis '('
framework/src/st_things/things_stack/things_stack.c:127: ERROR: [SPC_M_SEP] space prohibited after that open parenthesis '('
framework/src/st_things/things_stack/things_stack.c:130: ERROR: [SPC_M_SEP] space prohibited after that open parenthesis '('
framework/src/st_things/things_stack/things_stack.c:207: ERROR: [SPC_M_SEP] space prohibited after that open parenthesis '('
framework/src/st_things/things_stack/things_stack.c:212: ERROR: [SPC_M_SEP] space prohibited after that open parenthesis '('
framework/src/st_things/things_stack/things_stack.c:225: ERROR: [SPC_M_SEP] space prohibited after that open parenthesis '('
framework/src/st_things/things_stack/things_stack.c:230: ERROR: [SPC_M_SEP] space prohibited after that open parenthesis '('

**Comment by seinfra** [comment]
Target : [edfac5a7b3b2c78266f185e22ba2094fe13f3d13] - Code Rule Check (C++) OK.

File: framework/src/st_things/st_things.c [hunk 2]
```diff
@@ -449,6 +455,16 @@ int st_things_reset(void)
 	return ST_THINGS_ERROR_NONE;
 }
 
+int st_things_stop(void)
+{
+	if (things_stop_stack()) {
```
Comment (JeongJunSik) [inline framework/src/st_things/st_things.c]: What happens if Stop is called again after it is already stopped?

**Comment by seinfra** [comment]
Target : [edfac5a7b3b2c78266f185e22ba2094fe13f3d13] -

**Pull request closed without merging**